pkg/apis/core/validation/validation.go

```diff
@@ -5939,8 +5939,8 @@ func ValidateSecurityContext(sc *core.SecurityContext, fldPath *field.Path) fiel
 
 		if sc.Capabilities != nil {
 			for _, cap := range sc.Capabilities.Add {
-				if string(cap) == "CAP_SYS_ADMIN" {
```

Comment:
This is a good catch, but is a validation change that could break previously persisted data.

I'd expect this to leave the existing check in place and conditionally add the new check, if-and-only-if this is validating a create or an update of an object which didn't previously break the SYS_ADMIN rule

Reply:
Is the fix to disallow new objects (and invalid updates to already valid objects) with any capability that has `SYS_ADMIN` as a suffix when `allowPrivilegeEscalation=false`?

Reply:
yes, it makes sense, I also would like to know @tallclair opinion.

Reply:
> 

@enj this verification just rejects pod with allowPrivilegeEscalation: false and CAP_SYS_ADMIN,
frankly I was wander why SYS_ADMIN, but not SYS_MODULE, since in case of loading kernel module we can forcibly rewrite kernel structures and change no_new_priv attribute.

Reply:
> I'd expect this to leave the existing check in place and conditionally add the new check, if-and-only-if this is validating a create or an update of an object which didn't previously break the SYS_ADMIN rule

This bit is still outstanding

Reply:
> frankly I was wander why SYS_ADMIN, but not SYS_MODULE, since in case of loading kernel module we can forcibly rewrite kernel structures and change no_new_priv attribute.

Yeah, I agree with that. This is an arbitrary little bit of security policy that slipped into validation, and I'm not convinced it adds much.